internal/functions/efgs/download-batch.go

```diff
@@ -118,31 +121,27 @@ func downloadAndSaveKeys(ctx context.Context, config *downloadConfig) error {
 			return err
 		}
 
-		logger.Infof("Successfully enqueued downloaded keys for import to our Key server")
-
-		if err = updateDownloadedCounters(ctx, config.RealtimeDBClient, keysCount); err != nil {
-			logger.Warnf("Could not update EFGS download counters: %v", err)
-		}
-	}
+		logger.Infof("Successfully enqueued %v downloaded keys for import to our Key server", keysCount)
 
-	// Save params for next run:
+		// Save params for next run:
 
-	nextBatch := nextBatchParams(ctx, batchParams)
-	logger.Debugf("Next batch will be: %+v", nextBatch)
-	bytes, err := json.Marshal(nextBatch)
-	if err != nil {
-		return err
-	}
+		nextBatch := nextBatchParams(ctx, now, batchParams)
+		logger.Debugf("Next batch will be: %+v", nextBatch)
+		bytes, err := json.Marshal(nextBatch)
+		if err != nil {
+			return err
+		}
```

Comment:
the indent here looks weird, doesn't it?

Reply:
The indentation change is caused by moving this piece of code **inside** an if block, so far it was behind. Which was wrong...

Reply:
aha!